crates/datasources/src/object_store/http.rs

```diff
@@ -140,3 +172,232 @@ impl ObjStoreAccess for HttpStoreAccess {
             .await?)
     }
 }
+
+#[derive(Debug)]
+struct SimpleHttpStore {
+    url: Url,
+    // Used when content length available.
+    obj_store: HttpStore,
+}
+
+impl SimpleHttpStore {
+    async fn simple_get_req(&self, location: ObjectStorePath) -> Result<GetResult> {
+        let res = reqwest::get(self.url.clone()).await?;
+        if !res.status().is_success() {
+            return Err(ObjectStoreSourceError::InvalidHttpStatus(format!(
+                "getting data for '{}' resulted in error status: {}",
+                self.url,
+                res.status(),
+            )));
+        }
+
+        let download_file_dir = tempdir()?;
+
+        let download_file_path = download_file_dir.path().join(format!("{}", Uuid::new_v4()));
+
+        let mut contents = res.bytes_stream();
+
+        let mut download_file = tokio::fs::OpenOptions::new()
+            .create_new(true)
+            .write(true)
+            .read(true)
+            .open(&download_file_path)
+            .await?;
+
+        // We unfortunately need to download the contents first to get the
+        // content length since it is unknown.
+        while let Some(b) = contents.next().await {
+            let b = b?;
+            download_file.write_all(&b).await?;
+        }
+
+        let file_meta = download_file.metadata().await?;
+        let size = file_meta.len() as usize;
+
+        // Seek the downloaded file to start (before reading).
+        download_file.seek(SeekFrom::Start(0)).await?;
+
+        let stream = async_stream::stream! {
+            // Delete when the stream is dropped.
```

Comment:
don't we need to call this with move? I've been tending to use the `futures::stream::once()` rather than the macro to make this more clear, but if it's in an `Arc` then I think it ends up working ok?

Reply:
The macro expands to `async move`. Moreover, the compiler would just error if it didn't here.